File: basic recursion/c++/reverse_array.cpp
#include <iostream>
using namespace std;

void reverseArray(int arr[], int start, int end)
{
    if (start >= end)
    {
        return;
    }
    swap(arr[start], arr[end]);
    reverseArray(arr, start + 1, end - 1);
}

int main()
{
    int n;
    cout << "Enter the number of elements in the array: ";
    cin >> n;
    int *arr = new int[n];
    cout << "Enter the elements of the array: ";
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    reverseArray(arr, 0, n - 1);
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    return 0;
}

File: basic recursion/javascript/reverse_array.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

function reverseArray(arr, start, end) {
    if (start >= end) {
        return;
    }
    [arr[start], arr[end]] = [arr[end], arr[start]];
    reverseArray(arr, start + 1, end - 1);
}

rl.question("Enter the elements of the array separated by spaces: ", function(input) {
    const arr = input.split(" ");
    reverseArray(arr, 0, arr.length - 1);
    console.log("Reversed array:", arr.join(" "));
    rl.close();
});